### DISCUSSION TITLE
How to avoid stucking of test

### DISCUSSION THREAD

**Discussion by jojurko** [getgauge/taiko#2675] [General]
I am using  taiko: Version: 1.3.2 (Chromium: 100.0.4874.0)

During testing, scenario was passed. The same scenario as part of bigger specification, test is stuck in step, which is implemented in taiko with typescript:
@Step("Click button <txt>")
  public async clickButtonWithText(txt: string): Promise<void> {
    try {
       await waitFor(async () => await button(txt).isVisible(), 30000); //in this step is stuck
       await click(button(txt));   
    } catch (err) {
      Gauge.writeMessage(`Button ${txt} was not visible`);
    }
}

How I can avoid freezing test?

**Comment by marques-work** [comment]
You might need to do some more debugging to see on which expression it is hanging?

Perhaps `button(txt).isVisible()` is failing and hanging up to 30 seconds as you have configured in `waitFor()`?

It's possible that `button(txt)` is not getting the correct element (maybe there are several elements that match)? Or when run as part of a bigger suite, there are side effects from other tests that cause this element not to become visible. It's hard to say from here since we cannot see your test run, and that's only something you can do.

But I would step through and debug to test these hypotheses first.

**Comment by jojurko** [comment]
I was debugging it. my problem is not, that button is not visible, but the test is freezing in this step and not continue. I expected that after 30s test will continue, in case button is not visible by catch error, but this did not happen. 
During debugging I finished in line await waitFor(async () => await button(txt).isVisible(), 30000); after returned from taiko code and in VS code F11 or F10 did not make anything and test freeze here, until I stopped it.  
But thank you for your proposal. 
Probably this is not problem caused by taiko.
